cmd/ddev/cmd/config.go

```diff
@@ -33,22 +39,43 @@ var ConfigCommand = &cobra.Command{
 
 		c, err := ddevapp.NewConfig(appRoot, provider)
 		if err != nil {
-			// If there is an error reading the config and the file exists, we're not sure
-			// how to proceed.
-			if c.ConfigExists() {
-				util.Failed("Could not read config: %v", err)
-			}
+			util.Failed("Could not create new config: %v", err)
 		}
 
 		// Set the provider value after load so we can ensure we use the passed in provider value
 		// for this configuration.
 		c.Provider = provider
+		c.Name = siteName
+		c.Docroot = docrootRelPath
 
-		err = c.Config()
-		if err != nil {
-			util.Failed("There was a problem configuring your application: %v\n", err)
-		}
+		if siteName == "" && docrootRelPath == "" && pantheonEnvironment == "" {
+			err = c.PromptForConfig()
+			if err != nil {
+				util.Failed("There was a problem configuring your application: %v\n", err)
+			}
+		} else {
+			// Handle the case where flags have been passed in and we config that way
+			appType, err := ddevapp.DetermineAppType(c.Docroot)
+			if err != nil {
+				fullPath, _ := filepath.Abs(c.Docroot)
+				util.Failed("Failed to determine app Type (drupal7/drupal8/wordpress), your docroot may be incorrect - looking in directory %v (full path: %v), err: %v", c.Docroot, fullPath, err)
+			}
+			// If we found an application type just set it and inform the user.
+			util.Success("Found a %s codebase at %s\n", c.AppType, filepath.Join(c.AppRoot, c.Docroot))
+			prov, _ := c.GetProvider()
+
+			c.AppType = appType
 
+			// But pantheon *does* validate "Name"
+			err = prov.ValidateField("Name", c.Name)
+			if err != nil {
+				util.Failed("Failed to validate sitename %v with ddev prov %v err: %v", c.Name, provider, err)
```

Comment:
Suggestion: "Failed to validate sitename %v[sitename] with %v[provider] provider: %v[err]"